test/check-machines-snapshots

```diff
@@ -469,6 +477,117 @@ class TestMachinesSnapshots(machineslib.VirtualMachinesCase):
+        self.assertEqual(getSnapshotXML("subVmTest1", "snap-1").find("memory").get("file"),
+                         snapshot_dir + "/new/subVmTest1.snap-1.save")
+
+        # Open dialog for second snapshot
+        b.click("#vm-subVmTest1-add-snapshot-button")
+        b.set_input_text("#snapshot-create-dialog-name", "snap-2")
+
+        # Check that default comes from the previous snapshot
+        b.wait_val("#snapshot-create-dialog-memory-location input", snapshot_dir + "/new")
+
+        # Use a different location so that we have two snapshots and
+        # can test the sorting function that determines the most
+        # recent one.
+
+        snapshot_dir2 = "/var/lib/libvirt2" if connection == "system" else "/home/admin/.local/share/libvirt/2"
+
+        b.set_input_text("#snapshot-create-dialog-memory-location input", snapshot_dir2)
+        b.wait_text("#snapshot-create-dialog-memory-location li button", f'Create "{snapshot_dir2}"')
+        b.click("#snapshot-create-dialog-memory-location li:first-child button")
+
+        # Wait a couple of seconds to guarantee a different
+        # "creationTime" from the previous snapshot and create it
+        time.sleep(3)
+        b.click(".pf-v5-c-modal-box__footer button:contains(Create)")
+        b.wait_not_present("#vm-subVmTest1-create-snapshot-modal")
+
+        b.wait_text("#vm-subVmTest1-snapshot-0-name", "snap-2")
+        self.assertEqual(getSnapshotXML("subVmTest1", "snap-2").find("memory").get("file"),
+                         os.path.join(snapshot_dir2, "subVmTest1.snap-2.save"))
+
+        # Open dialog for third snapshot
+        b.click("#vm-subVmTest1-add-snapshot-button")
+        b.set_input_text("#snapshot-create-dialog-name", "snap-3")
+
+        # Check that default comes from the previous snapshot
+        b.wait_val("#snapshot-create-dialog-memory-location input", snapshot_dir2)
+
+        dev = self.add_ram_disk(50)
+        m.execute(f"mkfs.ext4 {dev}; mkdir -p /var/small; mount {dev} /var/small")
+        if connection == "session":
+            m.execute("chown admin:admin /var/small")
+        b.set_input_text("#snapshot-create-dialog-memory-location input", "/var/small")
+        b.wait_text("#snapshot-create-dialog-memory-location li:first-child button", "/var/small/")
+        b.click("#snapshot-create-dialog-memory-location li:first-child button")
+        b.wait_in_text("#snapshot-create-dialog-memory-location .pf-v5-c-helper-text__item.pf-m-warning",
+                       "Total space available: ")
+
+        # Go ahead and let it fail
+        b.click(".pf-v5-c-modal-box__footer button:contains(Create)")
+        b.wait_in_text("#vm-subVmTest1-create-snapshot-modal .pf-v5-c-alert",
+                       "No space left on device")
+        b.click(".pf-v5-c-modal-box__footer button:contains(Cancel)")
+        b.wait_not_present("#vm-subVmTest1-create-snapshot-modal")
+
+    def testMemoryLocationSession(self):
```

Comment:
What about testing the system session?

Reply:
Hmm, the `testMemoryLocation` test runs twice, once for the "system" connection and once for the "session" connection. Is there also a "system session" connection?

Reply:
Ah this is just an "unusual" way of doing it, I suppose which confused me, in podman we just introduce a helper like `_testMemoryLocation()` anyway not a blocker